docs: `enscli` and `ensskills` docs (#2112)
namehash/ensnode: docs/ensnode.io/src/content/docs/docs/integrate/integration-options/enscli.mdx

---
title: enscli
description: Coming soon — an agent- and human-friendly CLI for the ENS Omnigraph API, wrapping enssdk.
---

import { Aside, LinkCard } from "@astrojs/starlight/components";

<Aside type="caution" title="Coming soon">

> **`enscli`** is a planned ENS CLI that wraps [`enssdk`](/docs/integrate/integration-options/enssdk) to bring the ENS Omnigraph to the terminal. The npm name is reserved; we're still shaping the design and no commands ship yet.

Reviewer: ```suggestion
**`enscli`** is a planned ENS CLI that wraps [`enssdk`](/docs/integrate/integration-options/enssdk) to bring the ENS Omnigraph to the terminal. The npm name is reserved; we're still shaping the design. `enscli` will ship soon.
```

</Aside>

`enscli` will be the terminal-shaped entry point to the [ENS Omnigraph](/docs/integrate/omnigraph) — a single CLI for resolving names, looking up records, searching domains, and running ad-hoc queries against any ENSNode instance.

Designed to feel natural whether you're driving it yourself or letting an AI agent drive. From a terminal it's one `npx` away with sensible defaults against the public Omnigraph; for agents it's predictable arguments, structured output, and machine-readable help.

## Built for

- Developers exploring or validating an ENS integration from a terminal, without writing a script first.
- Operators wiring ENS lookups into shell pipelines, cron, or CI.
- AI coding agents driving [`ensskills`](/docs/integrate/integration-options/ensskills), which reach into the protocol through `enscli`.

<Aside type="note" title="Looking for `ensdb-cli`?">

> `ensdb-cli` is a separate planned tool — an operator-focused CLI for managing [ENSDb](/docs/services/ensdb) instances, snapshots, and schemas. `enscli` is for *querying* ENS; `ensdb-cli` is for *operating* the database that ENS is indexed into. See [ENSDb snapshots & CLI](/docs/integrate/ensdb-snapshots-and-cli).

Reviewer: @Goader Can you please take the lead on making it so that we can merge your work on PR 2112 now and then if you can push a commit onto Krzysztof's PR 2107 that will add then make this link get all wired up correctly.
